Tasks/Common/azure-arm-rest-v2/azure-arm-app-service.ts

```diff
@@ -159,6 +159,73 @@ export class AzureAppService {
         }
     }
 
+    public async swapSlotWithPreview(slotName: string, preserveVNet?: boolean): Promise<void> {
+        try {
+            var webRequest = new webClient.WebRequest();
+            webRequest.method = 'POST';
+            webRequest.body = JSON.stringify({
+                targetSlot: slotName,
+                preserveVnet: preserveVNet
+            });
+
+            var slotUrl: string = !!this._slot ? `/slots/${this._slot}` : '';
+            webRequest.uri = this._client.getRequestUri(`//subscriptions/{subscriptionId}/resourceGroups/{ResourceGroupName}/providers/Microsoft.Web/sites/{name}/${slotUrl}/applySlotConfig`, {
+            '{ResourceGroupName}': this._resourceGroup,
+            '{name}': this._name,
+            '{slotUrl}': slotUrl
+            }, null, '2016-08-01');
+
+            console.log(tl.loc('SwappingAppServiceSlotSlotsPhase1', this._name, this.getSlot(), slotName));
+            var response = await this._client.beginRequest(webRequest);
+            if(response.statusCode == 202) {
+                response= await this._client.getLongRunningOperationResult(response);
+            }
+            
+            if(response.statusCode != 200) {
+                throw ToError(response);
+            }
+
+            console.log(tl.loc('SwappedAppServiceSlotSlotsPhase1', this._name, this.getSlot(), slotName));
+            console.log(tl.loc('PreviewSwapPhase1', this._name, this.getSlot()));
+        }
+        catch(error) {
+            throw Error(tl.loc('FailedToSwapAppServiceSlotSlotsPhase1', this._name, this.getSlot(), slotName, this._client.getFormattedError(error)));
+        }
+    }
+
+    public async swapSlotWithPreviewCancel(slotName: string, preserveVNet?: boolean): Promise<void> {
+        try {
+            var webRequest = new webClient.WebRequest();
+            webRequest.method = 'POST';
+            webRequest.body = JSON.stringify({
+                targetSlot: slotName,
+                preserveVnet: preserveVNet
+            });
+
+            var slotUrl: string = !!this._slot ? `/slots/${this._slot}` : '';
+            webRequest.uri = this._client.getRequestUri(`//subscriptions/{subscriptionId}/resourceGroups/{ResourceGroupName}/providers/Microsoft.Web/sites/{name}/${slotUrl}/resetSlotConfig`, {
+            '{ResourceGroupName}': this._resourceGroup,
+            '{name}': this._name,
+            '{slotUrl}': slotUrl
+            }, null, '2016-08-01');
+
+            console.log(tl.loc('CancelSwapAppServiceSlotSlotsPhase1', this._name, this.getSlot(), slotName));
+            var response = await this._client.beginRequest(webRequest);
+            if(response.statusCode == 202) {
```

Comment:
This API doesnt return a polling url in its response. You should check just for status code 200,